rskj-core/src/test/java/co/rsk/peg/federation/PendingFederationTest.java

```diff
@@ -137,96 +137,153 @@ void testToString() {
+        FederationArgs federationArgs = new FederationArgs(federationMembers, creationTime, FEDERATION_CREATION_BLOCK_NUMBER, networkParams);
+        List<BtcECKey> erpPubKeys = federationConstants.getErpFedPubKeysList();
+        long activationDelay = federationConstants.getErpFedActivationDelay();
+        Federation expectedFederation = FederationFactory.buildNonStandardErpFederation(federationArgs, erpPubKeys, activationDelay, activations);
+
+        // Act
+        PendingFederation otherPendingFederation = new PendingFederation(
+            federationMembers
         );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
+            FEDERATION_CREATION_BLOCK_NUMBER,
+            federationConstants,
+            activations
+        );
+
+        // Assert
+        assertEquals(expectedFederation, builtFederation);
     }
 
-    @Test
-    void buildFederation_erp_ok_after_RSKIP_201_activation_after_RSKIP284_mainnet() {
-        testBuildFederation(
-            true,
-            true,
-            false,
-            NetworkParameters.ID_MAINNET,
-            6
+    @ParameterizedTest
+    @MethodSource("networkParameters")
+    void buildFederation_with6Members_afterRSKIP284Activations_shouldBuildNonStandardERPFed(NetworkParameters networkParams, FederationConstants federationConstants) {
+        // Arrange
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.hop400().forBlock(FEDERATION_CREATION_BLOCK_NUMBER);
+        FederationArgs federationArgs = new FederationArgs(federationMembers, creationTime, FEDERATION_CREATION_BLOCK_NUMBER, networkParams);
+        List<BtcECKey> erpPubKeys = federationConstants.getErpFedPubKeysList();
+        long activationDelay = federationConstants.getErpFedActivationDelay();
+        Federation expectedFederation = FederationFactory.buildNonStandardErpFederation(federationArgs, erpPubKeys, activationDelay, activations);
+
+        // Act
+        PendingFederation otherPendingFederation = new PendingFederation(
+            federationMembers
+        );
+        Federation builtFederation = otherPendingFederation.buildFederation(
+            creationTime,
+            FEDERATION_CREATION_BLOCK_NUMBER,
+            federationConstants,
+            activations
         );
+
+        // Assert
+        assertEquals(expectedFederation, builtFederation);
     }
 
-    @Test
-    void buildFederation_after_RSKIP_353_activation_testnet() {
-        testBuildFederation(
-            true,
-            true,
-            true,
-            NetworkParameters.ID_TESTNET,
-            6
+    @ParameterizedTest
+    @MethodSource("networkParameters")
+    void buildFederation_with6Members_afterRSKIP353Activation_BeforeRSKIP305Activations_shouldBuildP2SHERPFed(NetworkParameters networkParams, FederationConstants federationConstants) {
```

Comment:
We are missing a test case or assertion checking before and after [RSKIP293](https://github.com/rsksmart/rskj/blob/08b28ebe921216b8691119e285254860b9cbe20e/rskj-core/src/main/java/co/rsk/peg/bitcoin/NonStandardErpRedeemScriptBuilderFactory.java#L18) activation.

Reply:
I checked that RSKIP293 and RSKIP284 are activated together in Hop400. If both are activated the federation is NonStandardErpRedeemScriptBuilder. If none is activated (beforeRSKIP284), depending on the network, the federation is NonStandardErpRedeemScriptBuilderHardcoded or NonStandardErpRedeemScriptBuilderWithCsvUnsignedBE.

So the test https://github.com/rsksmart/rskj/blob/8184b656b37b6383d9576f82201559fe562ce124/rskj-core/src/test/java/co/rsk/peg/federation/PendingFederationTest.java#L188 already tests that the federation built is any of those.